test/integration/targets/cloudfront_distribution/tasks/main.yml

```diff
@@ -0,0 +1,361 @@
+- block:
+  - name: set yaml anchor
+    set_fact:
+      aws_connection_info: &aws_connection_info
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+    no_log: yes
+
+  - name: create cloudfront distribution using defaults
```

Comment:
Having tests for this is wonderful, thanks for adding this. I'm working on adding permissions/cleanup to CI for this PR, but this task fails with
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_nk_UzU/ansible_module_cloudfront_distribution.py", line 1179, in create_distribution
    return client.create_distribution_with_tags(DistributionConfigWithTags=distribution_config_with_tags)['Distribution']
  File "/Users/shertel/Workspace/ansible/venv/python2.7/lib/python2.7/site-packages/botocore/client.py", line 312, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/shertel/Workspace/ansible/venv/python2.7/lib/python2.7/site-packages/botocore/client.py", line 605, in _make_api_call
    raise error_class(parsed_response, operation_name)
InvalidArgument: An error occurred (InvalidArgument) when calling the CreateDistributionWithTags operation: The parameter CNAME contains one or more parameters that are not valid.

fatal: [localhost]: FAILED! => {
    "changed": false,
    "error": {
        "code": "InvalidArgument",
        "message": "The parameter CNAME contains one or more parameters that are not valid.",
        "type": "Sender"
    },
    "invocation": {
        "module_args": {
            "alias": "ansible-test-shertel-OSX-2-2369800001.github.io",
            "aliases": [
                "ansible-test-shertel-OSX-2-2369800001.github.io"
            ],
            "aws_access_key": "ASIAJ5DRFDZSUCUBOJAA",
            "aws_secret_key": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "cache_behaviors": null,
            "caller_reference": null,
            "comment": null,
            "custom_error_responses": null,
            "default_cache_behavior": {
                "compress": false,
                "default_t_t_l": 86400,
                "forwarded_values": {
                    "cookies": {
                        "forward": "none"
                    },
                    "headers": {
                        "quantity": 0
                    },
                    "query_string": true,
                    "query_string_cache_keys": {
                        "quantity": 0
                    }
                },
                "lambda_function_associations": {
                    "quantity": 0
                },
                "max_t_t_l": 31536000,
                "min_t_t_l": 0,
                "smooth_streaming": false,
                "target_origin_id": "ansible-test-shertel-OSX-2-2369800001-origin.example.com",
                "trusted_signers": {
                    "enabled": false,
                    "quantity": 0
                },
                "viewer_protocol_policy": "allow-all"
            },
            "default_origin_domain_name": null,
            "default_origin_path": null,
            "default_root_object": null,
            "distribution_id": null,
            "e_tag": null,
            "ec2_url": null,
            "enabled": null,
            "http_version": null,
            "ipv6_enabled": null,
            "logging": null,
            "origins": [
                {
                    "custom_headers": {
                        "quantity": 0
                    },
                    "custom_origin_config": {
                        "h_t_t_p_port": 80,
                        "h_t_t_p_s_port": 443,
                        "origin_keepalive_timeout": 5,
                        "origin_protocol_policy": "match-viewer",
                        "origin_read_timeout": 30,
                        "origin_ssl_protocols": {
                            "items": [
                                "TLSv1",
                                "TLSv1.1",
                                "TLSv1.2"
                            ],
                            "quantity": 3
                        }
                    },
                    "domain_name": "ansible-test-shertel-OSX-2-2369800001-origin.example.com",
                    "id": "ansible-test-shertel-OSX-2-2369800001-origin.example.com",
                    "origin_path": ""
                }
            ],
            "price_class": null,
            "profile": null,
            "purge_aliases": false,
            "purge_cache_behaviors": false,
            "purge_custom_error_responses": false,
            "purge_origins": true,
            "purge_tags": false,
            "region": null,
            "restrictions": null,
            "security_token": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "state": "present",
            "tags": {},
            "validate_certs": true,
            "viewer_certificate": null,
            "wait": false,
            "wait_timeout": 1800,
            "web_acl_id": null
        }
    },
    "msg": "Error creating distribution: An error occurred (InvalidArgument) when calling the CreateDistributionWithTags operation: The parameter CNAME contains one or more parameters that are not valid.",
    "response_metadata": {
        "http_headers": {
            "connection": "close",
            "content-length": "320",
            "content-type": "text/xml",
            "date": "Tue, 21 Nov 2017 15:58:21 GMT",
            "x-amzn-requestid": "cc4238b2-ced4-11e7-aecc-a983e62f2c9e"
        },
        "http_status_code": 400,
        "request_id": "cc4238b2-ced4-11e7-aecc-a983e62f2c9e",
        "retry_attempts": 0
    }
}
```

Reply:
Fixed this by lowercasing the cloudfront_hostname and resource_prefix since there could be uppercase letters which are not allowed.

Reply:
Tests pass locally, though they take forever.